client/my-sites/stats/activity-log-tasklist/index.jsx

```diff
@@ -0,0 +1,414 @@
+
+class ActivityLogTasklist extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		plugins: PropTypes.arrayOf( PropTypes.object ), // Plugins updated and those with pending updates
+
+		// Connected props
+		siteName: PropTypes.string.isRequired,
+		// Plugins already updated + those with pending updates.
+		// This extends plugins with the plugin update status.
+		pluginsWithUpdate: PropTypes.object.isRequired,
+
+		// Localize
+		translate: PropTypes.func.isRequired,
+		updateSinglePlugin: PropTypes.func.isRequired,
+		showErrorNotice: PropTypes.func.isRequired,
+		showInfoNotice: PropTypes.func.isRequired,
+		showSuccessNotice: PropTypes.func.isRequired,
+	};
+
+	state = {
+		dismissedPlugins: [],
+		queuedPlugins: [],
+	};
+
+	/**
+	 * Adds a single or multiple plugin slugs to a list of dismissed plugins.
+	 * If it receives a string, it assumes it's a valid plugin slug and adds it to the dismissed list.
+	 * When it doesn't receive a string, it adds all the plugin slugs to the dismissed list.
+	 *
+	 * @param {string|void} pluginSlug Slug of a plugin or nothing.
+	 */
+	dismissPlugins = pluginSlug => {
+		// ToDo: this should update some record in the tasklist API
+		const { pluginsWithUpdate, trackDismissPlugin, trackDismissPluginAll } = this.props;
+		let plugins;
+
+		if ( 'string' === typeof pluginSlug ) {
+			plugins = [ pluginSlug ];
+			trackDismissPlugin( pluginSlug );
+		} else {
+			plugins = Object.keys( pluginsWithUpdate );
+			trackDismissPluginAll();
+		}
+
+		this.setState( {
+			dismissedPlugins: union( this.state.dismissedPlugins, plugins ),
+		} );
+	};
+
+	/**
+	 * Goes to general plugin management screen.
+	 *
+	 * @returns {object} Action to redirect to plugins management.
+	 */
+	goManagePlugins = () => this.props.goManagePlugins( this.props.siteSlug );
+
+	/**
+	 * Goes to single plugin management screen.
+	 *
+	 * @param {string} pluginSlug Plugin slug, like "hello-dolly".
+	 *
+	 * @returns {object} Action to redirect to plugin management.
+	 */
+	goToPlugin = pluginSlug => this.props.goToPlugin( pluginSlug, this.props.siteSlug );
+
+	/**
+	 * Checks if the plugin update queue has more items and none is currently updating.
+	 * If so, updates the next plugin.
+	 */
+	continueQueue = () => {
+		if (
+			0 < this.state.queuedPlugins.length &&
+			! isPluginUpdating( Object.values( this.props.pluginsWithUpdate ) )
+		) {
+			this.updatePlugin( this.state.queuedPlugins[ 0 ] );
+		}
+	};
+
+	/**
+	 * Add a plugin to the update queue.
+	 *
+	 * @param {object} plugin Plugin to enqueue.
+	 *
+	 * @returns {undefined}
+	 */
+	enqueuePlugin = plugin =>
+		this.setState(
+			{
+				queuedPlugins: union( this.state.queuedPlugins, [ plugin ] ),
+			},
+			this.continueQueue
+		);
+
+	/**
+	 * Remove a plugin from the update queue.
+	 *
+	 * @returns {undefined}
+	 */
+	dequeuePlugin = () =>
+		this.setState(
+			{
+				queuedPlugins: drop( this.state.queuedPlugins ),
```

Comment:
it's fine to use but there's no need to pull in `lodash` here. the normative form in straight-up JS is…

```js
this.state.queuedPlugins.slice( 1 )
```

Reply:
Fantastic 👍